server/aws-lsp-codewhisperer/src/language-server/codeWhispererServer.ts

```diff
@@ -526,26 +515,31 @@ export const CodewhispererServerFactory =
             emitUserTriggerDecisionTelemetry(telemetry, session, timeSinceLastUserModification)
         }
 
-        const updateConfiguration = async () =>
-            lsp.workspace
-                .getConfiguration('aws.codeWhisperer')
-                .then(config => {
-                    if (config && config['includeSuggestionsWithCodeReferences'] === true) {
-                        includeSuggestionsWithCodeReferences = true
-                        logging.log('Configuration updated to include suggestions with code references')
-                    } else {
-                        includeSuggestionsWithCodeReferences = false
-                        logging.log('Configuration updated to exclude suggestions with code references')
-                    }
-                    if (config && config['shareCodeWhispererContentWithAWS'] === true) {
-                        codeWhispererService.shareCodeWhispererContentWithAWS = true
-                        logging.log('Configuration updated to share Amazon Q content with AWS')
-                    } else {
-                        codeWhispererService.shareCodeWhispererContentWithAWS = false
-                        logging.log('Configuration updated to not share Amazon Q content with AWS')
-                    }
-                })
-                .catch(reason => logging.log(`Error in GetConfiguration: ${reason}`))
+        const updateConfiguration = async () => {
+            try {
+                const qConfig = await lsp.workspace.getConfiguration(Q_CONFIGURATION_SECTION)
```

Comment:
I think here we just need to make sure we send the request with the correct parameters. According to LSP spec, https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#configurationParams an array of items has to be sent.

Reply:
I checked locally and it seems internally it somehow sends the request in the expected correct format. Can ignore this comment.

Reply:
FYI, under the hood `getConfiguration` is handled by this implementation, which converts arguments to LSP params 
https://github.com/microsoft/vscode-languageserver-node/blob/f58f4dff16ad2760028bb1cb95e882de30a1000f/server/src/common/configuration.ts#L4

Reply:
thanks for sharing the link Viktor.